be/src/io/cache/fs_file_cache_storage.cpp

```diff
@@ -588,6 +588,114 @@ void FSFileCacheStorage::load_cache_info_into_memory(BlockFileCache* _mgr) const
+
+                if (!cell || cell->is_deleted) {
+                    inconsistent_cache_context.emplace_back(
+                            hash, expiration_time, offset, cache_type,
+                            InconsistentCacheContext::InconsistentType::FILES_INCONSISTENT);
+                    continue;
+                }
+
+                size_t expected_size =
+                        (is_tmp && cell->file_block->state() == FileBlock::State::DOWNLOADING)
+                                ? cell->dowloading_size()
+                                : cell->size();
+                InconsistentCacheContext::InconsistentType inconsistent_type;
+                if (size != expected_size) {
+                    inconsistent_type |=
+                            InconsistentCacheContext::InconsistentType::SIZE_INCONSISTENT;
+                }
+                if (cache_type != cell->file_block->cache_type()) {
+                    inconsistent_type |=
+                            InconsistentCacheContext::InconsistentType::CACHE_TYPE_INCONSISTENT;
+                }
+                if (expiration_time != cell->file_block->expiration_time()) {
+                    inconsistent_type |= InconsistentCacheContext::InconsistentType::
+                            EXPIRATION_TIME_INCONSISTENT;
+                }
+
+                if (inconsistent_type != InconsistentCacheContext::InconsistentType::NONE) {
+                    inconsistent_cache_context.emplace_back(hash, expiration_time, offset,
+                                                            cache_type, inconsistent_type);
+                }
+            }
+        }
+    };
+    if constexpr (USE_CACHE_VERSION2) {
+        std::filesystem::directory_iterator key_prefix_it {_cache_base_path, ec};
+        if (ec) {
+            LOG(WARNING) << ec.message();
+            return;
+        }
+        for (; key_prefix_it != std::filesystem::directory_iterator(); ++key_prefix_it) {
+            if (!key_prefix_it->is_directory()) {
+                // skip version file
+                continue;
+            }
+            if (key_prefix_it->path().filename().native().size() != KEY_PREFIX_LENGTH) {
+                LOG(WARNING) << "Unknown directory " << key_prefix_it->path().native()
+                             << ", try to remove it";
+                std::error_code ec;
+                std::filesystem::remove(key_prefix_it->path(), ec);
+                if (ec) {
+                    LOG(WARNING) << "failed to remove=" << key_prefix_it->path()
+                                 << " msg=" << ec.message();
+                }
+                continue;
+            }
+            std::filesystem::directory_iterator key_it {key_prefix_it->path(), ec};
+            if (ec) {
+                LOG(WARNING) << ec.message();
+                continue;
+            }
+            check(key_it);
+        }
+    } else {
+        std::filesystem::directory_iterator key_it {_cache_base_path, ec};
```

Comment:
are you sure this check works for both VERSION1 and VERSION2? if not, just warning here "only support version2"